docs/static/azure-module.asciidoc

```diff
@@ -0,0 +1,591 @@
+requires more memory. 
+
+[source,ruby]
+----
+azure_event_hubs {
+   event_hub_connections => ["Endpoint=sb://example1...;EntityPath=event_hub_name1"]
+   max_batch_size => 125
+}
+----
+
+[id="plugins-{type}s-{plugin}-storage_connection"]
+===== `storage_connection`
+* Value type is <<string,string>>
+* No default value 
+
+Connection string for blob account storage. Blob account storage persists the
+offsets between restarts, and ensures that multiple instances of Logstash
+process different partitions. 
+When this value is set, restarts resume where processing left off.
+When this value is not set, the `initial_position` value is used on every restart.
+
+We strongly recommend that you define this value for production environments.
+
+
+[source,ruby]
+----
+azure_event_hubs {
+   event_hub_connections => ["Endpoint=sb://example1...;EntityPath=event_hub_name1"]
+   storage_connection => "DefaultEndpointsProtocol=https;AccountName=example...."
+}
+----
+
+[id="plugins-{type}s-{plugin}-storage_container"]
+===== `storage_container`
+* Value type is <<string,string>>
+* Defaults to the Event Hub name if not defined
+
+Name of the storage container used to persist offsets and allow multiple instances of {ls}
+to work together. 
+ 
+[source,ruby]
+----
+azure_event_hubs {
+   event_hub_connections => ["Endpoint=sb://example1...;EntityPath=event_hub_name1"]
+   storage_connection => "DefaultEndpointsProtocol=https;AccountName=example...."
+   storage_container => "my_container"
+}
+----
+
+To avoid overwriting offsets, you can use different storage containers. This is
+particularly important if you are monitoring two Event Hubs with the same name.
+You can use the advanced configuration model to configure different storage
+containers. 
+
+[source,ruby]
+----
+azure_event_hubs {
+     config_mode => "advanced"
+     consumer_group => "logstash"
+     storage_connection => "DefaultEndpointsProtocol=https;AccountName=example...."
+     event_hubs => [
+        {"insights-operational-logs" => {
+         event_hub_connection => "Endpoint=sb://example1..."
+         storage_container => "insights-operational-logs-1"
+        }},
+        {"insights-operational-logs" => {
+         event_hub_connection => "Endpoint=sb://example2..."
+         storage_container => "insights-operational-logs-2"
+        }}
+     ]
+   }
+----
+
+[id="plugins-{type}s-{plugin}-threads"]
+===== `threads`
+* Value type is <<number,number>>
+* Minimum value is `2`
+* Default value is `4`
+  
+Total number of threads used to process events. The value you set here applies
+to all Event Hubs. Even with advanced configuration, this value is a global
+setting, and can't be set per event hub. 
+
+[source,ruby]
+----
+azure_event_hubs {
+   threads => 4
+}
+----
+
+include::shared-module-options.asciidoc[]
+    
+[[run-azure]]
+==== Start the module 
+
+Run this command from the Logstash install directory:
```

Comment:
Run this command from the Logstash install directory (with the logstash.yml [properly configured](link)) ?? ...or something like that.

Reply:
@jakelandis  These are fantastic comments, and the updates are going well.  Link ^^ isn't working for me.  Can you point me in the right direction, please?